src/openloco/interop/interop.hpp

```diff
@@ -130,58 +130,180 @@ namespace openloco::interop
+        {
+            addr<TAddress, T>() |= v;
+            return *this;
+        }
+
+        loco_global& operator&=(const_reference v)
+        {
+            addr<TAddress, T>() &= v;
+            return *this;
+        }
+
+        loco_global& operator-=(const_reference v)
+        {
+            addr<TAddress, T>() -= v;
+            return *this;
+        }
+
+        loco_global& operator++()
+        {
+            addr<TAddress, T>()++;
+            return *this;
+        }
+
+        loco_global operator++(int)
+        {
+            addr<TAddress, T>()++;
+            return *this;
+        }
+
+        loco_global& operator--()
+        {
+            addr<TAddress, T>()--;
+            return *this;
+        }
+
+        loco_global operator--(int)
+        {
+            addr<TAddress, T>()--;
+            return *this;
+        }
+
+        reference operator*()
         {
             return addr<TAddress, T>();
         }
 
-        T* operator->()
+        pointer operator->()
         {
             return &(addr<TAddress, T>());
         }
+
+        constexpr size_t size() const
+        {
+            return sizeof(T);
+        }
     };
 
-    template<typename T, size_t TSize, uint32_t TAddress>
-    struct loco_global_array
+    template<typename T>
+    struct loco_global_iterator
     {
-        operator T*()
+    private:
+        T* _ptr;
+
+    public:
+        loco_global_iterator(T* p)
+            : _ptr(p)
+        {
+        }
+        loco_global_iterator& operator++()
+        {
+            ++_ptr;
+            return *this;
+        }
+        loco_global_iterator operator++(int)
+        {
+            loco_global_iterator<T> temp = *this;
+            ++_ptr;
+            return temp;
+        }
+        loco_global_iterator& operator--()
+        {
+            --_ptr;
+            return *this;
+        }
+        loco_global_iterator operator--(int)
+        {
+            loco_global_iterator<T> tmp = *this;
+            --_ptr;
+            return tmp;
+        }
+        bool operator==(const loco_global_iterator& rhs)
+        {
+            return _ptr == rhs._ptr;
+        }
+        bool operator!=(const loco_global_iterator& rhs)
+        {
+            return _ptr != rhs._ptr;
+        }
+        T& operator*()
+        {
+            return *_ptr;
+        }
+    };
+
+    template<typename T, size_t TCount, uintptr_t TAddress>
+    struct loco_global<T[TCount], TAddress>
+    {
+        typedef T type;
+        typedef type* pointer;
+        typedef type& reference;
+        typedef const type& const_reference;
+        typedef loco_global_iterator<T> iterator;
+
+        operator pointer()
         {
             return get();
         }
 
-        T& operator[](int idx)
+        pointer get() const
+        {
+            return reinterpret_cast<pointer>(&addr<TAddress, type>());
+        }
+
+        reference operator[](size_t idx)
```

Comment:
`loco_global::operator[]` needs to take an `int` parameter (as the old `loco_global_array` implementation did), not a `size_t`. Otherwise, the compiler gets upset, because the built-in `operator[]` for arrays has `int` as the indexing parameter. Specifically, you get this when you attempt to use `operator[]` on a `loco_global<T[N]>`:
```
error C2666: 'openloco::interop::loco_global<openloco::industry [128],6047068>::operator []': 2 overloads have similar conversions
note: could be 'openloco::industry &openloco::interop::loco_global<openloco::industry [128],6047068>::operator [](::size_t)'
note: or       'built-in C++ operator[(openloco::industry *, int)'
note: while trying to match the argument list '(openloco::interop::loco_global<openloco::industry [128],6047068>, openloco::industry_id_t)'
```

Also, when changing the type of `idx` back to `int`, the `if` statement condition for the bounds check should be restored to the way it was before in `loco_global_array::operator[]`: by first checking for negative values and then explicitly casting to `size_t` for the comparison with `size()`, it avoids signed/unsigned comparison problems as well as compiler warnings related thereto.